SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -277,51 +279,21 @@ bool USpatialActorChannel::ReplicateActor()
 		ActorReplicator->RepLayout->MergeChangeList((uint8*)Actor, HistoryItem.Changed, Temp, RepChanged);
 	}
 
-	const bool bCompareIndexSame = ActorReplicator->RepState->LastCompareIndex == ChangelistState->CompareIndex;
 	ActorReplicator->RepState->LastCompareIndex = ChangelistState->CompareIndex;
 
 	// Update the handover property change list.
-	USpatialTypeBinding* Binding = Interop->GetTypeBindingByClass(Actor->GetClass());
-	TArray<uint16> HandoverChanged;
-	if (Binding)
-	{
-		uint32 ShadowDataOffset = 0;
-		for (auto& PropertyInfo : Binding->GetHandoverHandlePropertyMap())
-		{
-			ShadowDataOffset = Align(ShadowDataOffset, PropertyInfo.Value.Property->GetMinAlignment());
-
-			const uint8* Data = PropertyInfo.Value.GetPropertyData((uint8*)Actor);
-
-			// Compare and assign.
-			if (RepFlags.bNetInitial || !PropertyInfo.Value.Property->Identical(HandoverPropertyShadowData.GetData() + ShadowDataOffset, Data))
-			{
-				HandoverChanged.Add(PropertyInfo.Key);
-				PropertyInfo.Value.Property->CopyCompleteValue(HandoverPropertyShadowData.GetData() + ShadowDataOffset, Data);
-			}
-			ShadowDataOffset += PropertyInfo.Value.Property->GetSize();
-		}
-	}
-
-	// We can skip the core actor if there are no new changelists to send, and we are not creating a new entity.
-	bool bReplicateCoreActor = true;
-	if (!bCreatingNewEntity && HandoverChanged.Num() == 0)
-	{
-		if (bCompareIndexSame || ActorReplicator->RepState->LastChangelistIndex == ChangelistState->HistoryEnd)
-		{
-			UpdateChangelistHistory(ActorReplicator->RepState);
-			bReplicateCoreActor = false;
-		}
-	}
+	FHandoverChangeState HandoverChangeState = GetHandoverChangeList(*ActorHandoverShadowData, Actor);
 
 	//todo-giray: We currently don't take replication of custom delta properties into account here because it doesn't use changelists.
 	// see ActorReplicator->ReplicateCustomDeltaProperties().
 
 	// If any properties have changed, send a component update.
-	if (bReplicateCoreActor && (RepFlags.bNetInitial || RepChanged.Num() > 0 || HandoverChanged.Num() > 0))
+	if (bCreatingNewEntity || RepChanged.Num() > 0 || HandoverChangeState.Num() > 0)
 	{		
-		if (RepFlags.bNetInitial && bCreatingNewEntity)
+		if (bCreatingNewEntity)
 		{
-			check(!Actor->IsFullNameStableForNetworking() || Interop->CanSpawnReplicatedStablyNamedActors());
+			// SAHIL: CHECK IF YOU PROPERLY FIX THIS
```

Comment:
Do we have a ticket? could we adjust the comment to link to the jira instead with explanation of the issue?

Reply:
I think this can be removed.